api/utils.py

```diff
@@ -0,0 +1,89 @@
+import logging
+from functools import partial
+
+import jwt
+from django.conf import settings
+from django.core.exceptions import (
+    ImproperlyConfigured,
+    PermissionDenied,
+)
+
+logger = logging.getLogger(__name__)
+
+# ----------------------------------------------------------------------------------------------------------------------
+# JWT Authentication
+# https://github.com/Aalto-LeTech/django-lti-login/blob/jwt-api/django_lti_login/api_views.py
```

Comment:
This time, don't include this reference, as I will remove that branch once the feature is merged in lti login. Normally, it's good style to reference the source :)

Reply:
I kind of would like to include the course path in the url. e.g. `/api/v1/def_course/upload`, which would make the jwt_auth part a bit nicer :)